cluster_vcf_records/vcf_file_read.py

```diff
@@ -1,13 +1,37 @@
+import gzip
 import itertools
 import operator
 import logging
 import multiprocessing
-
-import pyfastaq
+import os
 
 from cluster_vcf_records import vcf_record
 
 
+def open_vcf_file_for_reading(filename):
+    if filename.endswith(".gz"):
+        try:
+            return gzip.open(filename, "rt")
+        except:
+            raise OSError(
+                f"Error opening gzip file '{filename}'. Cannot continue"
+            )
+    elif filename.endswith(".bcf"):
+        try:
+            return os.popen('bcftools view ' + filename)
```

Comment:
You could just pass all files through this call as bcftools will read all forms and it will do it correctly (i.e. not based on the file extension)?

Reply:
I'd prefer not to because the code that reads the VCF is less strict than bcftools.

Reply:
Why is that an issue?

Reply:
People come up with all sorts of VCFs that are perfectly fine to use even though they don't conform to the spec. If we put everything through bcftools then we're more likely to have tools (particularly minos and gramtools) crashing.

Reply:
In my experience bcftools only fails if you break the specs. It gives lots of warning for things like undefined headers, but it will still work.

Reply:
ok, cool, if you want to try it out then fine by me. Could try to stop the user from seeing those warnings though? Can imagine things like "no contig header lines" warnings being spewed to stderr. I'd want to check that the tests still pass for this repo, minos, gramtools, varifier, and triphecta.